feature/experimental/policy/policy_vrf_selection/otg_tests/base_vrf_selection/base_vrf_selection_test.go

```diff
@@ -241,79 +247,109 @@ func applyForwardingPolicy(t *testing.T, ate *ondatra.ATEDevice, ingressPort str
 	gnmi.Replace(t, dut, intfConfPath.Config(), intf)
 
 	// Restart Protocols after policy change
-	ate.Topology().New().StopProtocols(t)
+	ate.OTG().StopProtocols(t)
 	time.Sleep(sleepOnChange)
-	ate.Topology().New().StartProtocols(t)
+	ate.OTG().StartProtocols(t)
 	time.Sleep(sleepOnChange)
 }
 
-func configureATE(t *testing.T, ate *ondatra.ATEDevice) []*ondatra.Flow {
-	t.Logf("*** Configuring ATE interfaces ...")
-	topo := ate.Topology().New()
+func configureATE(t *testing.T, ate *ondatra.ATEDevice) []gosnappi.Flow {
+	t.Logf("*** Configuring OTG interfaces ...")
+	topo := ate.OTG().NewConfig(t)
 
 	p1 := ate.Port(t, "port1")
 	p2 := ate.Port(t, "port2")
 
-	i1 := topo.AddInterface(ateSrc.Name).WithPort(p1)
-	i1.IPv4().WithAddress(ateSrc.IPv4CIDR()).WithDefaultGateway(dutSrc.IPv4)
-	i1.IPv6().WithAddress(ateSrc.IPv6CIDR()).WithDefaultGateway(dutSrc.IPv6)
-
-	i2 := topo.AddInterface(ateDst.Name).WithPort(p2)
-	i2.Ethernet().WithVLANID(vlan10)
-	i2.IPv4().WithAddress(ateDst.IPv4CIDR()).WithDefaultGateway(dutDst.IPv4)
-	i2.IPv6().WithAddress(ateDst.IPv6CIDR()).WithDefaultGateway(dutDst.IPv6)
-
-	i3 := topo.AddInterface(ateDst2.Name).WithPort(p2)
-	i3.Ethernet().WithVLANID(vlan20)
-	i3.IPv4().WithAddress(ateDst2.IPv4CIDR()).WithDefaultGateway(dutDst2.IPv4)
-	i3.IPv6().WithAddress(ateDst2.IPv6CIDR()).WithDefaultGateway(dutDst2.IPv6)
-
-	ethHeader := ondatra.NewEthernetHeader()
-	ipv4Header := ondatra.NewIPv4Header()
-	ipipHeader := ondatra.NewIPv4Header()
-	ipv6Header := ondatra.NewIPv6Header()
-	ipipDscp46Header := ondatra.NewIPv4Header().WithDSCP(46)
-	ipipDscp42Header := ondatra.NewIPv4Header().WithDSCP(42)
+	topo.Ports().Add().SetName(p1.ID())
+	srcDev := topo.Devices().Add().SetName(ateSrc.Name)
+	ethSrc := srcDev.Ethernets().Add().SetName(ateSrc.Name + ".Eth")
+	ethSrc.SetPortName(p1.ID()).SetMac(ateSrc.MAC)
+	ethSrc.Ipv4Addresses().Add().SetName(srcDev.Name() + ".IPv4").SetAddress(ateSrc.IPv4).SetGateway(dutSrc.IPv4).SetPrefix(int32(ateSrc.IPv4Len))
+	ethSrc.Ipv6Addresses().Add().SetName(srcDev.Name() + ".IPv6").SetAddress(ateSrc.IPv6).SetGateway(dutSrc.IPv6).SetPrefix(int32(ateSrc.IPv6Len))
+
+	topo.Ports().Add().SetName(p2.ID())
+	dstDev := topo.Devices().Add().SetName(ateDst.Name)
+	eth := dstDev.Ethernets().Add().SetName(ateDst.Name + ".Eth")
+	eth.SetPortName(p2.ID()).SetMac(ateDst.MAC)
+	eth.Vlans().Add().SetName(dstDev.Name() + "-VLAN").SetId(int32(vlan10))
+	eth.Ipv4Addresses().Add().SetName(dstDev.Name() + ".IPv4").SetAddress(ateDst.IPv4).SetGateway(dutDst.IPv4).SetPrefix(int32(ateDst.IPv4Len))
+	eth.Ipv6Addresses().Add().SetName(dstDev.Name() + ".IPv6").SetAddress(ateDst.IPv6).SetGateway(dutDst.IPv6).SetPrefix(int32(ateDst.IPv6Len))
+
+	dstDev2 := topo.Devices().Add().SetName(ateDst2.Name)
+	eth2 := dstDev2.Ethernets().Add().SetName(ateDst2.Name + ".Eth")
+	eth2.SetPortName(p2.ID()).SetMac(ateDst2.MAC)
+	eth2.Vlans().Add().SetName(dstDev2.Name() + "-VLAN").SetId(int32(vlan20))
```

Comment:
Could move this to a separate line below this block to make the test-specific config more clear.

Reply:
done